fix(ui): additional step for SOAP service and port

zregvart opened the pull request:
We used to have two screens on the "Imported Operations" step, akka
review step of the custom API client wizard. That caused issues with
error handling: the screen to specify service and port would throw an
`Error` if the provided WSDL file cannot be parsed. So in #9633 instead
of reaching this screen a notification would appear indicating of this
error situation. Unfortunately, this also changed the flow for the
OpenAPI. So when an error was generated editing the successfully parsed
OpenAPI document we would render the first screen (upload) of the wizard
and show the error notification. And in process loose any changes user
has made to the document.

This instead adds an additional step in the wizard "Additional
Configuration" to specify service/port for WSDL documents, and is
skipped for OpenAPI documents, also changing the order of the
"Imported Operations" (review) screen, so that it precedes this new
screen. This way we won't `Error`-out in trying to access nonexistent
data, as it could not be parsed.

Ref. https://issues.redhat.com/browse/ENTESB-16902

zregvart committed:
fix(ui): additional step for SOAP service and port

We used to have two screens on the "Imported Operations" step, akka
review step of the custom API client wizard. That caused issues with
error handling: the screen to specify service and port would throw an
`Error` if the provided WSDL file cannot be parsed. So in #9633 instead
of reaching this screen a notification would appear indicating of this
error situation. Unfortunately, this also changed the flow for the
OpenAPI. So when an error was generated editing the successfully parsed
OpenAPI document we would render the first screen (upload) of the wizard
and show the error notification. And in process loose any changes user
has made to the document.

This instead adds an additional step in the wizard "Additional
Configuration" to specify service/port for WSDL documents, and is
skipped for OpenAPI documents, also changing the order of the
"Imported Operations" (review) screen, so that it precedes this new
screen. This way we won't `Error`-out in trying to access nonexistent
data, as it could not be parsed.

Ref. https://issues.redhat.com/browse/ENTESB-16902

zregvart commented:
cc @gashcrumb, @mkralik3 this is what I think a _proper_ fix would do. It's quite big as you can see. It's also more work to adapt QE tests and probably needs adjusting the docs.
For `1.12.x` I can try to create a quick fix, one that would not be that _proper_. Or we can bite the bullet, up for discussion.

kahboom commented on app/ui-react/syndesis/src/modules/apiClientConnectors/locales/api-client-connectors-translations.en.json:
```diff
@@ -56,17 +24,14 @@
     "security": {
-      "address": "Address",
       "accessTokenUrl": "Access Token URL",
+      "address": "Address",
```
Happy to see it's not just me that appreciates alphabetical order 😆

zregvart replied:
Yeah I did that in #9652 to ease with OpenAPI JSON files copied between server and UI, but in general that (with time) will help us with merges/backports...

kahboom commented:
@zregvart - I mean, it's a lot of changes, but I do like the idea of moving the Service/Port config onto a separate page. I've gone through the changes and tested the branch locally, and it looks good, with a few small requests:

1. Please update the story filenames to match the step number; it just makes it _much_ easier to find them when you need to reference them. This probably also means adding another story for the new step/page.
2. Remove the footer in the `ApiConnectorCreatorService` (found in `packages/ui/src/Customization/apiClientConnectors/create`) since it's no longer needed, otherwise produces a duplicate Next button:

![Screen Shot 2021-07-02 at 11 56 08](https://user-images.githubusercontent.com/3844502/124271253-25de3000-db35-11eb-8632-c6d029641f55.png)

kahboom reviewed (changes requested):
@zregvart - I mean, it's a lot of changes, but I do like the idea of moving the Service/Port config onto a separate page. I've gone through the changes and tested the branch locally, and it looks good, with a few small requests:

1. Please update the story filenames to match the step number; it just makes it _much_ easier to find them when you need to reference them. This probably also means adding another story for the new step/page.
2. Remove the footer in the `ApiConnectorCreatorService` (found in `packages/ui/src/Customization/apiClientConnectors/create`) since it's no longer needed, otherwise produces a duplicate Next button:

![Screen Shot 2021-07-02 at 11 56 08](https://user-images.githubusercontent.com/3844502/124271253-25de3000-db35-11eb-8632-c6d029641f55.png)

zregvart commented:
Fat fingered that. Thanks @kahboom for the review, I'll get to fixing this.

zregvart committed:
refactor: support more than one action summary

Reducing action summary to a single summary doesn't suit SOAP client
connectors, as within a WSDL file there could be many services with
different operations, a contrast to OpenAPI documents where there could
be only one API service per document.

So instead of summing up all operations from all services we can now
have one action summary per service.

So when importing a WSDL file with multiple services we can now show the
number of actions along with the number of services. Further refactoring
might opt to show a number of actions per service along with the name of
the service, but lets consider this a first step.

And this is all so we can show the correct number of actions/services
and set the default serviceName/portName/address when the WSDL file
contains more than one service.

zregvart committed:
fix: keep track of the selected service and port

We're loosing the selected serviceName and portName when navigating the
SOAP client connector wizard. This makes sure that the selected values
are kept. Also removes the superfluous `Next` button on the select
service/port name screen ('Additional Configuration' screen).

zregvart committed:
fix: create connector with the selected address

We did not persist the selected address when creating the SOAP client
connector, we reverted to the default address set to the first service
in the WSDL file.

This makes sure that the address is passed in `configuredProperties`
when creating the connector.

For this an additional property was needed that contains a map of port
to address so that the correct address for the selected service/port
combination is presented on the last (details) screen of the wizard.

zregvart committed:
fix: don't throw an error if errors are returned

Throwing an exception here breaks the wizard and reverts to the upload-
-provide-url screen. This is especially awkward when user introduces an
error in the OpenAPI document by editing. In some (most) cases if the
OpenAPI document can be parsed by Apicurio Studio the error can be
corrected and user doesn't need to start from the beginning.

Most notably changing the `openapi` key makes the document unparsable
and it leads to loss of data. Some changes, like changing the `type` in
the object's JSON schema are recoverable.

We should be throwing `Error`s only in exceptional circumstances and
errors reported from the server are not that.

This changes the flow, so the notification of an error will not be shown
but rather the user will land on the first page of the wizard, which
allows the user to edit the OpenAPI document and resolve the issue.

zregvart commented:
Many new fixes were added for this, more details in the commit messages.

zregvart commented:
This also should fix https://issues.redhat.com/browse/ENTESB-17281 via a671e49

Delawen commented:
Rerunning jobs because one failed. Just in case.

zregvart commented:
> Rerunning jobs because one failed. Just in case.

There was a TSLint issue in the UI, should be okay now.
